core/modules/system/image.gd.inc

```diff
@@ -348,6 +348,15 @@ function image_gd_get_info(stdClass $image) {
 
   if (isset($data) && is_array($data)) {
     $extensions = array('1' => 'gif', '2' => 'jpg', '3' => 'png');
+    if (defined('IMAGETYPE_WEBP')) {
+      $gd_info = gd_info();
+      if (isset($gd_info['WebP Support']) && $gd_info['WebP Support'] == TRUE) {
+        // Extend mapping of constants to extensions, as the php and gd
+        // versions support webp.
+        // @see https://www.php.net/manual/en/image.constants.php
+        $extensions['18'] = 'webp';
```

Comment:
What is this "magic" number, and why do we set 1,2,3 for gif/jpg/png in the `$extensions` array a few lines above?

Reply:
Hehe, that's why I put in the link to the image constant definitions on php.net. "18" is the value for webp.